### DISCUSSION TITLE
IPXE fails to compile

### DISCUSSION THREAD

**Discussion by joda666** [ipxe/ipxe#470] [General]
[ZINFO] bin/15ad07b0.rom.zinfo
[ZBIN] bin/15ad07b0.rom.zbin
[FINISH] bin/15ad07b0.rom

To create a bootable floppy, type
cat bin/ipxe.dsk > /dev/fd0
where /dev/fd0 is your floppy drive. This will erase any
data already on the disk.

To create a bootable USB key, type
cat bin/ipxe.usb > /dev/sdX
where /dev/sdX is your USB key, and is not a real hard
disk on your system. This will erase any data already on
the USB key.

To create a bootable CD-ROM, burn the ISO image
bin/ipxe.iso to a blank CD-ROM.

These images contain drivers for all supported cards. You
can build more customised images, and ROM images, using
make bin/.

===========================================================
rm bin/10ec8139.rom.zinfo bin/version.ipxe.dsk.o bin/1af41000.rom.bin bin/ipxe.pxe.zinfo bin/version.undionly.kpxe.o bin/rtl8139.rom.zinfo bin/15ad07b0.rom.zbin bin/8086100f.mrom.zbin bin/rtl8139.rom.bin bin/version.1af41000.rom.o bin/808610d3.mrom.zbin bin/ipxe.hd.zbin bin/10222000.rom.bin bin/undionly.kpxe.zinfo bin/ipxe.dsk.zinfo bin/ipxe.hd.bin bin/80861209.rom.zbin bin/8086100f.mrom.zinfo bin/version.10500940.rom.o bin/8086100e.mrom.zbin bin/version.ipxe.hd.o bin/15ad07b0.rom.bin bin/version.15ad07b0.rom.o bin/808610d3.mrom.bin bin/ipxe.hd bin/ipxe.lkrn.zbin bin/version.808610d3.mrom.o bin/8086100e.mrom.zinfo bin/ipxe.hd.zinfo bin/ipxe.dsk.zbin bin/version.ipxe.pxe.o bin/8086100f.mrom.bin bin/rtl8139.rom.zbin bin/1af41000.rom.zinfo bin/8086100e.mrom.bin bin/10222000.rom.zbin bin/version.ipxe.lkrn.o bin/15ad07b0.rom.zinfo bin/808610d3.mrom.zinfo bin/version.10ec8139.rom.o bin/ipxe.pxe.zbin bin/version.10222000.rom.o bin/ipxe.lkrn.bin bin/ipxe.pxe.bin bin/undionly.kpxe.bin bin/ipxe.dsk.bin bin/version.8086100e.mrom.o bin/version.80861209.rom.o bin/10222000.rom.zinfo bin/10ec8139.rom.bin bin/80861209.rom.zinfo bin/ipxe.lkrn.zinfo bin/10ec8139.rom.zbin bin/10500940.rom.zbin bin/80861209.rom.bin bin/version.8086100f.mrom.o bin/10500940.rom.zinfo bin/10500940.rom.bin bin/usbdisk.bin bin/undionly.kpxe.zbin bin/version.rtl8139.rom.o bin/1af41000.rom.zbin
server3:/usr/local/ipxe/src #

End of compilation. Works fine.

Can't compile with embedded file:

make undionly_embed.kpxe EMBED=ipxe.embed
[VERSION] bin/version.undionly_embed.kpxe.o
[LD] bin/undionly_embed.kpxe.tmp
ld:--defsym:2: undefined symbol `obj_undionly_embed' referenced in expression
make: *** [Makefile.housekeeping:1220: bin/undionly_embed.kpxe.tmp] Fehler 1
rm bin/version.undionly_embed.kpxe.o

make undionly_embed.kpxe EMBED=ipxe.embed 
  [DEPS] crypto/certstore.c
  [DEPS] crypto/privkey.c
  [DEPS] image/embedded.c
make: *** Keine Regel, um „undionly_embed.kpxe“ zu erstellen.  Schluss.
No Rule to create ... Finish.

Whats wrong? In

**Comment by NiKiZe** [comment]
there is no target named `undionly_embed.kpxe`
You probably want `bin/undionly.kpxe` - you then copy that file to whatever name you want.
(just as it says: "Keine Regel, um „undionly_embed.kpxe“ zu erstellen. Schluss." )

Some information on alternative targets at: https://ipxe.org/appnote/buildtargets

**Comment by stappersg** [comment]
@joda666 your feedback is appricated.
And you even can earn bonus karma points by closing this issue.
